src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,112 +183,570 @@ def __repr__(self):
+        """
+        user_allocation = json.loads(json_string)
+        variant = user_allocation.get(FeatureFlagConstants.VARIANT, None)
+        users = user_allocation.get(FeatureFlagConstants.USERS, None)
+        if not variant or not users:
+            raise ValidationError(
+                "User variant allocation must contain required '%s' and '%s' attributes: \n%s"
+                % (FeatureFlagConstants.VARIANT, FeatureFlagConstants.USERS, json.dumps(user_allocation, indent=2, ensure_ascii=False))
+            )
+
+        return cls(variant=variant, users=users)
+
+    def __repr__(self):
+        feature_user_allocation = {
+            FeatureFlagConstants.VARIANT: self.variant,
+            FeatureFlagConstants.USERS: self.users,
+        }
+
+        return json.dumps(feature_user_allocation, indent=2, ensure_ascii=False)
+
+
+class FeatureGroupAllocation:
+    """
+    Feature group allocation class.
+
+    :ivar str variant:
+        The name of the variant to use
+        if the group allocation matches a group the current user is in.
+    :ivar list groups:
+        Collection of groups where if the current user is in any of these groups,
+        the variant specified in the group allocation is used.
+    """
+
+    def __init__(self, variant, groups):
+        self.variant = variant
+        self.groups = groups
+
+    @classmethod
+    def convert_from_json(cls, json_string):
+        """
+        Convert JSON string to FeatureGroupAllocation object
+
+        Args:
+            json_string - JSON string
+
+        Return:
+            FeatureGroupAllocation object
+        """
+        group_allocation = json.loads(json_string)
+        variant = group_allocation.get(FeatureFlagConstants.VARIANT, None)
+        groups = group_allocation.get(FeatureFlagConstants.GROUPS, None)
+        if not variant or not groups:
+            raise ValidationError(
+                "Group variant allocation must contain required '%s' and '%s' attributes: \n%s"
+                % (FeatureFlagConstants.VARIANT, FeatureFlagConstants.GROUPS, json.dumps(group_allocation, indent=2, ensure_ascii=False))
+            )
+
+        return cls(variant=variant, groups=groups)
+
+    def __repr__(self):
+        feature_group_allocation = {
+            FeatureFlagConstants.VARIANT: self.variant,
+            FeatureFlagConstants.GROUPS: self.groups,
+        }
+
+        return json.dumps(feature_group_allocation, indent=2, ensure_ascii=False)
+
+
+class FeaturePercentileAllocation:
+    """
+    Feature percentile allocation class.
+
+    :ivar str variant:
+        The name of the variant to use if the calculated percentile for the current user falls in the provided range.
+    :ivar number from:
+        The lower end of the percentage range for which this variant will be used.
+    :ivar number to:
+        The upper end of the percentage range for which this variant will be used.
+    """
+
+    def __init__(self, variant, From, to):
```

Comment:
```suggestion
    def __init__(self, variant, from_, to):
```

I would suggest adding a trailing underscore like we do with id [here](https://github.com/Azure/azure-cli/blob/a3dc72bc3bd881a57cd856e124362ac6e99178da/src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py#L64)

Reply:
Makes sense Updated